src/components/MessagesList/MessagesGroup/Message/MessagePart/Forwarder.vue

```diff
@@ -99,16 +100,37 @@ export default {
 		selectedConversationName() {
 			return this.$store.getters?.conversation(this.selectedConversationToken).name
 		},
+
+		/**
+		 * Object containing all the mentions in the message that will be forwarded
+		 * @returns {Object} mentions.
+		 */
+		mentions() {
+			const mentions = {}
+			for (const key in this.messageObject.messageParameters) {
+				if (key.startsWith('mention')) {
+					mentions[key] = this.messageObject.messageParameters[key]
+				}
+			}
+			return mentions
+		},
 	},
 
 	methods: {
 		async setSelectedConversationToken(token) {
 			this.selectedConversationToken = token
+			const messageToBeForwarded = cloneDeep(this.messageObject)
+			// Overwrite the selected conversation token
+			messageToBeForwarded.token = token
+			// If there are mentions in the message to be forwarded, replace them in the message
+			// text.
+			if (this.mentions !== {}) {
+				for (const mention in this.mentions) {
+					messageToBeForwarded.message = messageToBeForwarded.message.replace(`{${mention}}`, '@' + this.mentions[mention].name)
```

Comment:
```suggestion
					messageToBeForwarded.message = messageToBeForwarded.message.replace(`{${mention}}`, '@' + this.mentions[mention].id)
```
Id or really name?

Reply:
Always name, so that mentions for users will work and for rooms will display the room name instead of the token

Reply:
But mentioning with display name does not work if you name is not your id (mostly the case).

Reply:
Since you can not mention a room but only at-all we can replace it

But same issue is with guests. A guest mention in another room does not resolve

Reply:
it doesn't harm if the mention doesn't resolve, unless the user is expecting some different behavior, like auto-inviting ? but I doubt it